watch/example_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package watch
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/storage/inmem"
+)
+
+func ExampleWatcher_Query() {
+	ctx := context.Background()
+	store := inmem.NewFromObject(loadSmallTestData())
+
+	mod, err := ast.ParseModule("example", "package y\nr = s { s = data.a }")
+	if err != nil {
+		// Handle error
+	}
+
+	compiler := ast.NewCompiler()
+	if compiler.Compile(map[string]*ast.Module{"example": mod}); compiler.Failed() {
+		// Handle error
+	}
+
+	txn, err := store.NewTransaction(ctx, storage.WriteParams)
+	if err != nil {
+		// Handle error
+	}
+
+	// Create a new Watcher that uses the given store and compiler to monitor
+	// queries.
+	w, err := New(ctx, store, compiler, txn)
+	if err != nil {
+		// Handle error
+	}
+	if err := store.Commit(ctx, txn); err != nil {
+		// Handle error
+	}
+
+	// Create a new watch on the query. Whenever its result changes, the result of
+	// the change will be sent to `handle.C`.
+	handle, err := w.Query("x = data.y")
+	if err != nil {
+		// Handle error
+	}
+
+	// This kind of coordination between the store writing and the watch
```

Comment:
Nit: this comment buries the lede a bit; I would just include the second half, e.g.,

```
This example synchronizes the reader and writer to make the output deterministic.
```